contrib/node/src/python/pants/contrib/node/register.py

```diff
@@ -37,12 +39,19 @@ def build_file_aliases():
 
 
 def register_goals():
+  # Register descriptions.
+  Goal.register('node-install', 'Installs a node_module target.')
+
+  # Register tasks.
   task(name='node', action=NodeRepl).install('repl')
   task(name='node', action=NodeResolve).install('resolve')
+  # Because of execution order, resolve.node is always executed before node-resolve-local
+  # This means that installation occurs twice, once in the working dir and again in the local dir
   task(name='node', action=NodeRun).install('run')
   task(name='node', action=NodeBuild).install('compile', first=True)
   task(name='node', action=NodeTestTask).install('test')
   task(name='node', action=NodeBundleTask).install('bundle')
+  task(name='node', action=NodeInstall).install('node-install')
```

Comment:
A task with the same name as the goal is a little bit easier to use. See https://www.pantsbuild.org/dev_tasks.html#task-installation-associate-task-with-goals

To do that, install it as:
```
 task(name='node-install', action=NodeInstall).install()
```

Then rather than having its options scoped as `--node-install-node-*`, they will just be `--node-install-*`.